sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -640,6 +642,9 @@ struct _pi_context : _pi_object {
   // If AllowBatching is true, then the command list returned may already have
   // command in it, if AllowBatching is false, any open command lists that
   // already exist in Queue will be closed and executed.
+  // When using immediate commandlists, retrieves an immediate command list
+  // for executing on this device. Immediate commandlists are created only
+  // once per device and after that they are reused.
```

Comment:
> Immediate commandlists are created only once per device and after that they are reused.

They are currently created for each PI queue, and that indeed appears to be redundant. Can we address this?

Reply:
If they are not SYCL-queue specific, then waiting on one SYCL queue will cause a wait on all SYCL queues. Not sure we want that.

Reply:
You mean the barrier used to sync the queue will wait for submissions from all SYCL queues to this device's immediate command-list? We certainly don't want that and it may not even be legal, I think. Do you have info how multiple immediate-command lists to the same ordinal/index are going to inter-operate?

Reply:
We will need to clarify this with the L0 team. If we have two immediate commandlists that have the same ordinal and index, the implementation useful to us will be that a barrier on one will not affect the other.

Reply:
Right, please follow up on that. Meanwhile let's keep them per SYCL queue and not per SYCL device, so please update the comment.

Reply:
OK